UniversityApp/Controllers/StudentsController.cs

```diff
@@ -32,6 +33,17 @@ public async Task<IActionResult> Get()
             return Ok(result);
         }
 
+        [HttpGet("search")]
+        public async Task<IActionResult> Search([FromQuery] StudentSearchModel student, int pageNumber = 1, int pageSize = 10)
```

Comment:
StudentSearchModel ->StudentSearchDto

pageNumber and pageSize can also go into model. If many properties are in model I would suggest to pass them in body of the request